core/storage/question/gae_models.py

```diff
@@ -316,25 +316,58 @@ def get_question_skill_links_by_skill_ids(
         ) * question_count
 
         if not start_cursor == '':
-            cursor = datastore_query.Cursor(urlsafe=start_cursor)
-            question_skill_link_models, next_cursor, more = cls.query(
-                cls.skill_id.IN(skill_ids)
-                # Order by cls.key is needed alongside cls.last_updated so as to
-                # resolve conflicts, if any.
-                # Reference SO link: https://stackoverflow.com/q/12449197
-            ).order(-cls.last_updated, cls.key).fetch_page(
-                question_skill_count,
-                start_cursor=cursor
-            )
+            direction = start_cursor[-1]
+            random_id = start_cursor[-13:-1]
+            cursor = datastore_query.Cursor(urlsafe=start_cursor[:-13])
+            if direction == '1':
+                questions_list, next_cursor, more = cls.query(
+                    cls.skill_id.IN(skill_ids)
+                ).order(cls.question_id, cls.key).filter(
+                    cls.question_id > random_id
+                ).fetch_page(
+                    question_skill_count,
+                    start_cursor=cursor
+                )
+            else:
+                questions_list, next_cursor, more = cls.query(
+                    cls.skill_id.IN(skill_ids)
+                ).order(cls.question_id, cls.key).filter(
+                    cls.question_id <= random_id
+                ).fetch_page(
+                    question_skill_count,
+                    start_cursor=cursor
+                )
```

Comment:
You can replace this entire section with

```
questions_list, next_cursor, more = get_questions_list(skill_ids, random_id, question_skill_count, direction, cursor)
```

Reply:
Done.